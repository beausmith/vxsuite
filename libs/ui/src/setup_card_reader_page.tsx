import React, { useEffect } from 'react';
import { Main, MainChild } from './main';
import { Prose } from './prose';
import { Screen } from './screen';
import { fontSizeTheme } from './themes';

function doNothing() {
  // do nothing
}

interface Props {
  useEffectToggleLargeDisplay?: () => void;
  usePollWorkerLanguage?: boolean;
}

export function SetupCardReaderPage({
  useEffectToggleLargeDisplay = doNothing,
  usePollWorkerLanguage = true,
}: Props): JSX.Element {
  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(useEffectToggleLargeDisplay, []);

  const connectMessage = usePollWorkerLanguage
    ? 'Please ask a poll worker to connect card reader.'
    : 'Please connect the card reader to continue.';

  return (
    <Screen white>
      <Main>
        <MainChild center maxWidth={false}>
          <Prose textCenter maxWidth={false} theme={fontSizeTheme.large}>
            <h1>Card Reader Not Detected</h1>
            <p>{connectMessage}</p>
          </Prose>
        </MainChild>
      </Main>
    </Screen>
  );
}
