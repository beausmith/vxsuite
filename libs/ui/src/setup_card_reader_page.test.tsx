import React from 'react';
import { render } from '@testing-library/react';

import { SetupCardReaderPage } from './setup_card_reader_page';

describe('renders SetupCardReaderPage', () => {
  test('with no useEffect trigger as expected', () => {
    const { container } = render(<SetupCardReaderPage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('triggers useEffect property', () => {
    const triggerFn = jest.fn();
    const { container } = render(
      <SetupCardReaderPage useEffectToggleLargeDisplay={triggerFn} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(triggerFn).toHaveBeenCalled();
  });

  test('renders SetupCardReaderPage with usePollWorkerLanguage set to false', () => {
    const { container } = render(
      <SetupCardReaderPage usePollWorkerLanguage={false} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
