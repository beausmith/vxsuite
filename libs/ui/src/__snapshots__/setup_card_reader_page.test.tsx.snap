// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SetupCardReaderPage renders SetupCardReaderPage with usePollWorkerLanguage set to false 1`] = `
<div
  class="sc-dkPtyc ftEvXz"
>
  <main
    class="sc-bdvvaa jnfvZB"
  >
    <div
      class="sc-gsDJrp fCJDVu"
    >
      <h1>
        Card Reader Not Detected
      </h1>
      <p>
        Please connect the card reader to continue.
      </p>
    </div>
  </main>
</div>
`;

exports[`renders SetupCardReaderPage triggers useEffect property 1`] = `
<div
  class="sc-dkPtyc ftEvXz"
>
  <main
    class="sc-bdvvaa jnfvZB"
  >
    <div
      class="sc-gsDJrp fCJDVu"
    >
      <h1>
        Card Reader Not Detected
      </h1>
      <p>
        Please ask a poll worker to connect card reader.
      </p>
    </div>
  </main>
</div>
`;

exports[`renders SetupCardReaderPage with no useEffect trigger as expected 1`] = `
<div
  class="sc-dkPtyc ftEvXz"
>
  <main
    class="sc-bdvvaa jnfvZB"
  >
    <div
      class="sc-gsDJrp fCJDVu"
    >
      <h1>
        Card Reader Not Detected
      </h1>
      <p>
        Please ask a poll worker to connect card reader.
      </p>
    </div>
  </main>
</div>
`;
